File: sp48/crypt/prohorow/test1.pas
{****************************************************************************}
{*                                                                          *}
{*      Data Encryption ToolKit. Version 1.3                                *}
{*                                                                          *}
{*      test1.pas -- test module for DES module 1                           *}
{*                                                                          *}
{*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    *}
{*                                                                          *}
{*      Purpose:                                                            *}
{*        Test InitEncryption and EncryptBlock procedures.                  *}
{*                                                                          *}
{****************************************************************************}


program Test1;

  uses Encrypt;

  const
    TestValue: array [1..4] of Integer = (1, 2, 3, 4);
    Key: array [1..4] of Integer = (100, 200, 300, 400);
    EncryptionKeyNumber = 1;

{$i encrcon.inc}

  var
    KeyArea: array [1..EncryptionKeyAreaSize] of Char;

  procedure TestWrite;

    var
      I: Integer;

    begin
      for I := 1 to 4 do
        Write (TestValue[I], ' ');
      WriteLn
    end;

  begin
    TestWrite;
    InitEncryption (KeyArea, Key, EncryptionKeyNumber);
    EncryptBlock (KeyArea, TestValue);
    TestWrite;
    InitEncryption (KeyArea, Key, - EncryptionKeyNumber);
    EncryptBlock (KeyArea, TestValue);
    TestWrite
  end.

File: sp48/crypt/prohorow/test1.c
/****************************************************************************/
/*                                                                          */
/*      Data Encryption ToolKit. Version 1.3                                */
/*                                                                          */
/*      test1.c -- test module for DES module 1                             */
/*                                                                          */
/*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    */
/*                                                                          */
/*      Purpose:                                                            */
/*        Test init_encryption and encrypt_block functions.                 */
/*                                                                          */
/****************************************************************************/


#include <stdio.h>
#include "encrypt.h"

int test_value[4] = {1, 2, 3, 4};
int key[4] = {100, 200, 300, 400};
const int KEY_NUMBER = 1;
char key_area[ENCRYPTION_KEY_AREA_SIZE (KEY_NUMBER)];

void test_write () {
  for (int i = 0; i < 4; i ++)
    printf ("%i ", test_value[i]);
  printf ("\n");
  }

void main () {
  test_write ();
  init_encryption (key_area, key, KEY_NUMBER);
  encrypt_block (key_area, test_value);
  test_write ();
  init_encryption (key_area, key, - KEY_NUMBER);
  encrypt_block (key_area, test_value);
  test_write ();
  }
